File: IPWorks MQ Samples/AMQP/php/amqp.php
<?php
/*
 * IPWorks MQ 2024 PHP Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks MQ in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworksmq
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */
require_once('../include/ipworksmq_amqp.php');
require_once('../include/ipworksmq_const.php');
?>
<?php
global $amqp;
class MyAMQP extends IPWorksMQ_AMQP
{
  function fireSSLServerAuthentication($param) {
    $param['accept'] = true;
    return $param;
  }
  function fireMessageIn($param) {
    global $amqp;
    echo "Incoming message from <" . $param['linkname'] . ">: " . $amqp->getReceivedMessageValue() . "\n";
  }
}

function input($prompt) {
  echo $prompt;
  $handle = fopen("php://stdin", "r");
  $data = trim(fgets($handle));
  fclose($handle);
  return $data;
}

try {
  $amqp = new MyAMQP();
  $amqp->setContainerId("ContainerId");

  if ($argc < 4) {
    echo "Usage: php amqp.php [options] server user password -port port -u user -p password -ssl\n\n";
    echo "    -ssl       whether to use SSL\n";
    echo "    -port      the port of the AMQP server\n";
    echo "    server     ...\n";
    echo "    user       ...\n";
    echo "    password   ...\n";
    echo "Example: php amqp.php -port 5671 -ssl 127.0.0.1 username password\n\n";
    return;
  } else {
    $amqp->setRemoteHost($argv[$argc-3]);
    $amqp->setUser($argv[$argc-2]);
    $amqp->setPassword($argv[$argc-1]);
  }

  for ($i = 0; $i < $argc; $i++) {
    if (str_starts_with($argv[$i], "-")) {
      if ($argv[$i] == "-ssl") { $amqp->setSSLEnabled(true); }
      if ($argv[$i] == "-port") { $amqp->setRemotePort($argv[$i + 1]);}
    }
  }
  try {
    $amqp->doConnect();
  } catch (Exception $ex) {
    echo "Error connecting: " . $ex->getMessage() . "\n";
    return;
  }

  $amqp->doCreateSession("SessionId");

  try {
    $amqp->doCreateSenderLink("SessionId", "SenderLinkName", "TargetName");
  } catch (Exception $ex) {
    echo "Error creating sender link: " . $ex->getMessage() . "\n";
    return;
  }

  $amqp->setReceiveMode(1); // Retrieve
  $amqp->setRetrieveTimeout(5);

  try {
    $amqp->doCreateReceiverLink("SessionId", "ReceiverLinkName", "TargetName");
  } catch (Exception $ex) {
    echo "Error creating receiver link: " . $ex->getMessage() . "\n";
    return;
  }

  $command = "";
  while ($command != "q") {
    if ($command == "s") {
      $amqp->doResetMessage();
      $amqp->setMessageValueType(17); // string
      $amqp->setMessageValue(input("Enter message to send: "));
      try {
        $amqp->doSendMessage("SenderLinkName");
      } catch (Exception $ex) {
        "Error sending message: " . $ex->getMessage . "\n";
        return;
      }
      echo "Message sent!\n";
    } elseif ($command == "f") {
      echo "Fetching message...\n";
      try {
        $amqp->doRetrieveMessage("ReceiverLinkName");
      } catch (Exception $ex) {
        if (str_starts_with($ex->getMessage(), "201")) {
          echo "Timeout - no message received.\n";
        } else {
          echo "Error fetching message: " . $ex->getMessage() . "\n";
          return;
        }
      }
    }
    $command = input("Choose send message (s), fetch message (f), or quit (q): ");
  }
} catch (Exception $ex) {
  echo "Error: " . $ex->getMessage() . "\n";
}
?>

File: IPWorks MQ Samples/AMQP/java/amqp.java
/*
 * IPWorks MQ 2024 Java Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks MQ in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworksmq
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */

import java.io.*;
import ipworksmq.*;

public class amqp extends ConsoleDemo{

    public static void main(String[] args) {
        System.out.println("*******************************************************");
        System.out.println("* This is a demo for the IP*Works IoT AMQP Class.     *");
        System.out.println("* It allows simple message sending and receiving.     *");
        System.out.println("*******************************************************");
        AMQP amqp = new AMQP();
        
        
        try {
        //add listeners
        	amqp.addAMQPEventListener(new AMQPEventListener() {
    			
    			@Override
    			public void messageOutcome(AMQPMessageOutcomeEvent arg0) {
    				// TODO Auto-generated method stub
    				
    			}
    			
    			@Override
    			public void messageOut(AMQPMessageOutEvent arg0) {
    				// TODO Auto-generated method stub
    				
    			}
    			
    			@Override
    			public void messageIn(AMQPMessageInEvent arg0) {
    				System.out.println("\nMessage Received: "+amqp.getReceivedMessage().getValue());
    				
    			}
    			
    			@Override
    			public void log(AMQPLogEvent arg0) {
    				// TODO Auto-generated method stub
    				
    			}
    			
    			@Override
    			public void linkReadyToSend(AMQPLinkReadyToSendEvent arg0) {
    				// TODO Auto-generated method stub
    				
    			}
    			
    			@Override
    			public void error(AMQPErrorEvent arg0) {
    				// TODO Auto-generated method stub
    				
    			}
    			
    			@Override
    			public void disconnected(AMQPDisconnectedEvent arg0) {
    				if(arg0.statusCode!=0) {
						System.out.println("Error disconnecting: "+arg0.statusCode+" - "+arg0.description);
					}
    				System.exit(0);
    			}
    			
    			@Override
    			public void connectionStatus(AMQPConnectionStatusEvent arg0) {
    				// TODO Auto-generated method stub
    				
    			}
    			
    			@Override
    			public void connected(AMQPConnectedEvent arg0) {
    				System.out.println("Connected.");
    				
    			}
    			
    			@Override
    			public void SSLStatus(AMQPSSLStatusEvent arg0) {
    				// TODO Auto-generated method stub
    				
    			}
    			
    			@Override
    			public void SSLServerAuthentication(AMQPSSLServerAuthenticationEvent arg0) {
    				arg0.accept=true;
    				
    			}
    		});
        	
        //configure settings
        
        //set container id, ssl enabled and remote host
        amqp.setContainerId("ContainerId");
        boolean ssl = Boolean.valueOf(ask("Use SSL","?")=='y');
        amqp.setSSLEnabled(ssl);
        amqp.setRemoteHost(prompt("Remote Host"));
        
        //prompt for remote port
        String defPort = "5672";
        if(ssl) {
        	defPort = "5671"; 
    	}
        amqp.setRemotePort(Integer.valueOf(prompt("Remote Port",":",defPort)));
        
        //prompt for user and password
        amqp.setUser(prompt("User"));
        amqp.setPassword(prompt("Password"));
        
        //connect to host
        amqp.connectTo(amqp.getRemoteHost(), amqp.getRemotePort());
        
        //prompt for automatic or fetch flow
        char flow = ' ';
        while(flow!='a'&&flow!='f') {
    		flow = ask("Choose (a)utomatic or (f)etch flow",":","a/f");
        }
        if(flow=='f') {
        	amqp.setReceiveMode(AMQP.rmRetrieve);
            amqp.setRetrieveTimeout(5);
        }
        
        //set session id and create sender link with a unique link name and name of target node (at most once flow by default, settled = true)
        amqp.createSession("SessionId");
        amqp.createSenderLink("SessionId", "SenderLinkName", "TargetName");
        
        //create receiver link with existing session ID, unique link name, and name of the target node        
        amqp.createReceiverLink("SessionId", "ReceiverLinkName", "TargetName");
        
        char response = ask("\nWould you like to (s)end message, (f)etch message, or (q)uit","?","s/f/q");
        while(response!='q') {
        	if(response=='s') {
        		amqp.resetMessage();
        		amqp.getMessage().setValueType(17); //string
        		amqp.getMessage().setValue(prompt("Enter message"));
        		amqp.sendMessage("SenderLinkName");
        		System.out.println("Message sent.\n");
        	} else if(response=='f') {
        		System.out.println("Fetching message...");
        		try {
        		amqp.retrieveMessage("ReceiverLinkName");
        		} catch(IPWorksMQException e) {
        			if(e.getCode()==201) {
        				System.out.println("Timeout. No message received.");
        			} else {
        				displayError(e);
        			}
        		}
        	}
        	response = ask("Would you like to (s)end message, (f)etch message, or (q)uit","?","s/f/q");
        }
        	
        } catch (Exception e) {
            displayError(e);
        }
    }
}
class ConsoleDemo {
  private static BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));

  static String input() {
    try {
      return bf.readLine();
    } catch (IOException ioe) {
      return "";
    }
  }
  static char read() {
    return input().charAt(0);
  }

  static String prompt(String label) {
    return prompt(label, ":");
  }
  static String prompt(String label, String punctuation) {
    System.out.print(label + punctuation + " ");
    return input();
  }
  static String prompt(String label, String punctuation, String defaultVal) {
      System.out.print(label + " [" + defaultVal + "]" + punctuation + " ");
      String response = input();
      if (response.equals(""))
        return defaultVal;
      else
        return response;
  }

  static char ask(String label) {
    return ask(label, "?");
  }
  static char ask(String label, String punctuation) {
    return ask(label, punctuation, "(y/n)");
  }
  static char ask(String label, String punctuation, String answers) {
    System.out.print(label + punctuation + " " + answers + " ");
    return Character.toLowerCase(read());
  }

  static void displayError(Exception e) {
    System.out.print("Error");
    if (e instanceof IPWorksMQException) {
      System.out.print(" (" + ((IPWorksMQException) e).getCode() + ")");
    }
    System.out.println(": " + e.getMessage());
    e.printStackTrace();
  }

  /**
   * Takes a list of switch arguments or name-value arguments and turns it into a map.
   */
  static java.util.Map<String, String> parseArgs(String[] args) {
    java.util.Map<String, String> map = new java.util.HashMap<String, String>();
    
    for (int i = 0; i < args.length; i++) {
      // Add a key to the map for each argument.
      if (args[i].startsWith("-")) {
        // If the next argument does NOT start with a "-" then it is a value.
        if (i + 1 < args.length && !args[i + 1].startsWith("-")) {
          // Save the value and skip the next entry in the list of arguments.
          map.put(args[i].toLowerCase().replaceFirst("^-+", ""), args[i + 1]);
          i++;
        } else {
          // If the next argument starts with a "-", then we assume the current one is a switch.
          map.put(args[i].toLowerCase().replaceFirst("^-+", ""), "");
        }
      } else {
        // If the argument does not start with a "-", store the argument based on the index.
        map.put(Integer.toString(i), args[i].toLowerCase());
      }
    }
    return map;
  }
}




